core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -253,7 +261,8 @@ public void testSerialization() throws IOException {
         try (BytesStreamOutput output = new BytesStreamOutput()) {
             testQuery.writeTo(output);
             try (StreamInput in = StreamInput.wrap(output.bytes())) {
-                QB deserializedQuery = createEmptyQueryBuilder().readFrom(in);
+                QueryBuilder prototype = queryParserService.queryParser(testQuery.queryId()).getBuilderPrototype();
+                QueryBuilder deserializedQuery = prototype.readFrom(in);
```

Comment:
oh boy I didn't realize we could get rid of createEmptyQueryBuilder, this is great!